assets/js/modules/analytics/components/dashboard/AnalyticsAdSenseDashboardWidgetTopPagesTable.js

```diff
@@ -16,122 +16,100 @@
  * limitations under the License.
  */
 
-/**
- * External dependencies
- */
-import { map } from 'lodash';
-
 /**
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Component } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
+import Data from 'googlesitekit-data';
 import { getTimeInSeconds, numberFormat, getModulesData } from '../../../../util';
 import withData from '../../../../components/higherorder/withdata';
 import { TYPE_MODULES } from '../../../../components/data';
 import { getDataTableFromData, TableOverflowContainer } from '../../../../components/data-table';
-import Layout from '../../../../components/layout/layout';
 import PreviewTable from '../../../../components/preview-table';
 import ctaWrapper from '../../../../components/notifications/cta-wrapper';
 import AdSenseLinkCTA from '../common/AdSenseLinkCTA';
 import { analyticsAdsenseReportDataDefaults, isDataZeroForReporting } from '../../util';
-
-class AnalyticsAdSenseDashboardWidgetTopPagesTable extends Component {
-	static renderLayout( component ) {
-		const { accountURL } = getModulesData().adsense;
-		return (
-			<Layout
-				header
-				title={ __( 'Performance over previous 28 days', 'google-site-kit' ) }
-				headerCtaLabel={ __( 'Advanced Settings', 'google-site-kit' ) }
-				headerCtaLink={ accountURL }
-			>
-				{ component }
-			</Layout>
-		);
+import { STORE_NAME } from '../../datastore/constants';
+import AnalyticsAdSenseDashboardWidgetLayout from './AnalyticsAdSenseDashboardWidgetLayout';
+
+const { useSelect } = Data;
+
+const AnalyticsAdSenseDashboardWidgetTopPagesTable = ( { data } ) => {
+	const accountID = useSelect( ( select ) => select( STORE_NAME ).getAccountID() );
+	const profileID = useSelect( ( select ) => select( STORE_NAME ).getProfileID() );
+	const internalWebPropertyID = useSelect( ( select ) => select( STORE_NAME ).getInternalWebPropertyID() );
+
+	const adsenseDeepLink = useSelect( ( select ) => select( STORE_NAME ).getServiceURL(
+		{ path: `/report/content-pages/a${ accountID }w${ internalWebPropertyID }p${ profileID }/explorer-table.plotKeys=%5B%5D&_r.drilldown=analytics.pagePath:~2F` }
```

Comment:
Do the `plotKeys` parts of the URL require manually escaping as done here? I'd think it's better to escape the entire URL we get back from the selector rather than particular pieces manually.